src/main/java/com/ibm/as400/access/Permission.java

```diff
@@ -722,6 +721,23 @@ public Enumeration<String> getAuthorizedUsers()
       }
     }
 
+    /**
+     * Returns a List of authorized users.
+     * @return A list of authorized users.
+     *
+     **/
+    public List<String> getAuthorizedUsersList() {
+        synchronized (userPermissionsLock_) {
+            int count = userPermissions_.size();
+            List<String> names = new ArrayList<>();
+            for (int i = 0; i < count; i++) {
+                UserPermission userPermission = (UserPermission) userPermissions_.elementAt(i);
```

Comment:
Since `userPermissions_` is a vector object, it can be iterated more elegantly and efficiently with a Java [for-each loop](https://docs.oracle.com/javase/1.5.0/docs/guide/language/foreach.html). 
For instance:
```java
for(UserPermission userPermission: userPermissions) {
    names.add(userPermission.getUserID());
}

Reply:
updated it